guide/voice/audio-player.md

```diff
@@ -0,0 +1,124 @@
+# Audio Player
+
+Audio players can be used to play audio across voice connections. A single audio player can play the same audio over multiple voice connections.
+
+## Cheat sheet
+
+### Creation
+
+Creating an audio player is simple:
+
+```ts
+import { createAudioPlayer } from '@discordjs/voice';
+
+const player = createAudioPlayer();
+```
+
+You can also customise the behaviours of an audio player. For example, the default behaviour is to pause when there are no active subscribers for an audio player. This behaviour can be configured to either pause, stop, or just continue playing through the stream:
+
+```ts
+import { createAudioPlayer, NoSubscriberBehaviour } from '@discordjs/voice';
+
+const player = createAudioPlayer({
+	behaviours: {
+		noSubscriber: NoSubscriberBehaviour.Pause
+	}
+});
+```
+
+### Deletion
+
+If you no longer require an audio player, you can `stop()` it and then remove references to it so that it gets garbage collected.
+
+```ts
+connection.stop();
+// Make sure that you don't hold any references to the
+// connection to ensure it gets removed from memory
+```
+
+### Playing audio
+
+You can create [audio resources](./audio-resources) and then play them on an audio player.
+
+```ts
+const resource = createAudioResource('/home/user/track.mp3');
+player.play(resource);
+
+// Play "track.mp3" across two voice connections
+connection1.subscribe(player);
+connection2.subscribe(player);
+```
+
+::: warning
+**Audio players can play one audio resource at most.** If you try to play another audio resource while one is already playing on the same player, the existing one is destroyed and replaced with the new one.
```

Comment:
Is this the actual behaviour? If two players play from the same resource, and one changes to another, does it destroy the resource, leaving the other player to [UB](https://en.wikipedia.org/wiki/Undefined_behavior)?

If this is an intended limitation, it should then be documented, or even, redesigned to prevent this pattern from happening.

Reply:
At the moment, two players should **not** be able to share the same resource. This behaviour isn't currently enforced, but you raise a good point - I will make `.play()` throw an error if a resource is passed that is currently locked by some other player.

However, the original meaning is still the same. If you're playing resource A on an audio player, if you then play resource B, then resource A is destroyed and flushed, and then the audio player begins to consume resource B.

I think this behaviour is fine, since it works similarly to this in Discord.js v12. It would also be awkward to have to do `player.stop()` before being able to play a new resource.